Lib/test/pickletester.py

```diff
@@ -2125,6 +2168,37 @@ def remove_frames(pickled, keep_frame=None):
                             count_opcode(pickle.FRAME, pickled))
             self.assertEqual(obj, self.loads(some_frames_pickle))
 
+    def test_framed_write_sizes(self):
+        if not hasattr(self, 'pickler'):
```

Comment:
Could you explain what tests are skipped?

Reply:
Some test classes do not have a pickler attribute and just uses the `self.dumps` / `self.loads` methods. For those, we cannot change the `fast` flag to disable memoization.

Reply:
It seems to me that currently all test classes with the `test_framed_write_sizes` method have the `pickler` attribute.

The danger of skipping the test if not hasattr(self, 'pickler'), is that after changing the name of the `pickler` attribute **all** tests will be silently skipped.

Reply:
`test.test_pickletools.OptimizedPickleTests` does not have it.

Reply:
I have removed the skip to stay on the safe side. At the cost of running the test for `fast=False` twice.

Reply:
Hum this one has not been updated. Let me fix that.

Reply:
Actually, for this test I am not sure what to do: `OptimizedPickleTests` does not have `self.pickler` but this test only makes sense when the test case expose a `self.pickler` class.

Reply:
It uses `self.pickler` only in `self.pickler(w, proto).dump(small_objects)`. Why not use just `self.dump(small_objects)` here?

Reply:
Because the goal of this test is to measure the size of the chunks written into a custom writer object. This cannot be done with the `dumps` API.

Reply:
Ah, in this case I would skip the test in `OptimizedPickleTests`. But it is more reliable to set `test_framed_write_sizes = None` in `OptimizedPickleTests` instead of testing `hasattr(self, 'pickler')`. If the `pickler` attribute will be renamed this test will loudly fail instead of became silently skipped.